File: mobile/lib/features/operation_financiere/domain/entities/solde_periode.dart
/// Totaux de la carte solde de l'accueil pour une période, calculés côté
/// backend via `GET /operations-financieres/solde` (opérations annulées
/// exclues). Remplace le calcul qui était fait sur la liste locale des
/// opérations.
class SoldePeriode {
  final double revenus;
  final double depenses;
  final double solde;

  const SoldePeriode({
    required this.revenus,
    required this.depenses,
    required this.solde,
  });

  static const zero = SoldePeriode(revenus: 0, depenses: 0, solde: 0);

  factory SoldePeriode.fromJson(Map<String, dynamic> json) => SoldePeriode(
        revenus: (json['revenus'] as num? ?? 0).toDouble(),
        depenses: (json['depenses'] as num? ?? 0).toDouble(),
        solde: (json['solde'] as num? ?? 0).toDouble(),
      );
}

File: backend/src/main/java/com/tmk/vtcmanager/application/domain/operation/SoldePeriode.java
package com.tmk.vtcmanager.application.domain.operation;

import java.math.BigDecimal;

/**
 * Agrégat de la carte solde de l'écran d'accueil sur une période :
 * total des revenus, total des dépenses (opérations annulées exclues) et solde
 * net. Reproduit côté serveur le calcul jusqu'ici fait dans le mobile.
 */
public record SoldePeriode(BigDecimal revenus, BigDecimal depenses) {

    public SoldePeriode {
        revenus = revenus == null ? BigDecimal.ZERO : revenus;
        depenses = depenses == null ? BigDecimal.ZERO : depenses;
    }

    /** Solde net de la période : revenus − dépenses. */
    public BigDecimal solde() {
        return revenus.subtract(depenses);
    }
}
